src/Features/Core/Portable/SimplifyInterpolation/AbstractSimplifyInterpolationDiagnosticAnalyzer.cs

```diff
@@ -63,18 +64,16 @@ private void AnalyzeInterpolation(OperationAnalysisContext context)
                 return;
             }
 
-            var locations = ImmutableArray.Create(interpolation.Syntax.GetLocation());
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                UnnecessaryWithSuggestionDescriptor,
+                unnecessaryLocations.First(),
```

Comment:
do we know we always have at least one unnecessary location?

Reply:
Yes. Is there something I should do?

Reply:
nope. t his is fine.